### PR TITLE
API docs updates

### PR DESCRIPTION
**Pull request by kasenvr**

### CODE REVIEW

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR730/Vircadia-Alpha-PR730-ddbb920_bcd5f242.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR730/Vircadia-Alpha-PR730-ddbb920.tar_bcd5f242.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR730/Vircadia-Alpha-PR730-ddbb920.tar_bcd5f242.Z

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR730/Vircadia-Alpha-PR730-ddbb920_bcd5f242.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR730/Vircadia-Alpha-PR730-ddbb920_bcd5f242.exe

File: libraries/entities/src/EntityScriptingInterface.h [hunk 1]
```diff
@@ -110,9 +111,11 @@ class ParabolaToEntityIntersectionResult {
  * displayed and so knows about. For assignment client scripts, the entities available are those that are "seen" by the 
  * {@link EntityViewer}. For entity server scripts, all entities are available.</p>
  *
- * <h3>Entity Types</h3>
+ * <h3>Entity Types & Properties</h3>
  *
  * <p>For a list of the entity types that you can use, see {@link Entities.EntityType|Entity Types}.</p>
+ * 
+ * <p>While each entity type has properties that are specific to it, there are properties that are universal to all entity types, see {@link Entities.EntityProperties|Entity Properties}.</p>
```
Comment (ctrlaltdavid) [inline libraries/entities/src/EntityScriptingInterface.h:118]:
```suggestion
 * <p>For the properties of the different entity types, see {@link Entities.EntityProperties|Entity Properties}. Some properties are universal to all entity types, and some are specific to particular entity types.</p>
```

File: libraries/script-engine/src/ScriptEngine.h [hunk 2]
```diff
@@ -118,6 +118,17 @@ class EntityScriptDetails {
  *       <li><code>"agent"</code>: An assignment client script.</li>
  *     </ul>
  *     <em>Read-only.</em>
+ * @property {string} type - The type of engine that is currently in use:
```
Comment (ctrlaltdavid) [inline libraries/script-engine/src/ScriptEngine.h:121]:
```suggestion
 * @property {string} type - The type of script engine that is currently in use:
```

File: libraries/script-engine/src/ScriptEngine.h [hunk 3]
```diff
@@ -118,6 +118,17 @@ class EntityScriptDetails {
...
+ *     <ul>
+ *       <li><code>"client"</code>: The engine is an interface.</li>
+ *       <li><code>"entity_client"</code>: The engine is an entity simulated in an interface.</li>
+ *       <li><code>"entity_server"</code>: The engine is an entity simulated in a server.</li>
+ *       <li><code>"agent"</code>: The engine is an agent.</li>
+ *       <li><code>"avatar"</code>: The engine is an avatar.</li>
+ *     </ul>
```
Comment (ctrlaltdavid) [inline libraries/script-engine/src/ScriptEngine.h:128]: Hmmm. These are a little different from the script types listed at the top of the JSDoc pages: Interface Scripts • Client Entity Scripts • Avatar Scripts • Server Entity Scripts • Assignment Client Scripts ?
If so, add a note that they are different (and perhaps why / how)?

File: libraries/script-engine/src/ScriptEngine.h [hunk 4]
```diff
@@ -118,6 +118,17 @@ class EntityScriptDetails {
...
+ *     <em>Read-only.</em>
+ * @property {string} filename - The filename of the script's file.
```
Comment (ctrlaltdavid) [inline libraries/script-engine/src/ScriptEngine.h:130]:
```suggestion
 * @property {string} filename - The filename of the script file.
```
And path (i.e., "path and filename")?

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

File: libraries/script-engine/src/ScriptEngine.h (hunk 3 above)
Comment (kasenvr) [reply libraries/script-engine/src/ScriptEngine.h:128]: If I knew why... :D

File: libraries/entities/src/EntityScriptingInterface.h (hunk 1 above)
Comment (kasenvr) [reply libraries/entities/src/EntityScriptingInterface.h:118]: I feel like this is... incorrect? My link should be to the universal entity properties, however the text implies that you are going to a list of entity-type-specific properties.

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR730/Vircadia-Alpha-PR730-93a4268.tar_88fc2136.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR730/Vircadia-Alpha-PR730-93a4268_88fc2136.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR730/Vircadia-Alpha-PR730-93a4268.tar_88fc2136.gz

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR730/Vircadia-Alpha-PR730-93a4268_88fc2136.exe

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR730/Vircadia-Alpha-PR730-93a4268_88fc2136.dmg

File: libraries/entities/src/EntityScriptingInterface.h (hunk 1 above)
Comment (ctrlaltdavid) [reply libraries/entities/src/EntityScriptingInterface.h:118]: The table of common properties has a list of links to the entity-specific properties, under the table. The description of the table says this ...
```
Different entity types have different properties: some common to all entities (listed in the table) and some specific to each EntityType (linked to below).
```
I.e., the common properties table is a good way in to discover all entity properties.

Comment (kasenvr) [reply libraries/entities/src/EntityScriptingInterface.h:118]: Ah, alright.

File: libraries/script-engine/src/ScriptEngine.h [hunk 5]
```diff
@@ -118,6 +118,17 @@ class EntityScriptDetails {
...
  *     </ul>
  *     <em>Read-only.</em>
+ * @property {string} type - The type of script engine that is running the script:
+ *     <ul>
+ *       <li><code>"client"</code>: The engine is an interface.</li>
+ *       <li><code>"entity_client"</code>: The engine is an entity simulated in an interface.</li>
+ *       <li><code>"entity_server"</code>: The engine is an entity simulated in a server.</li>
+ *       <li><code>"agent"</code>: The engine is an agent.</li>
+ *       <li><code>"avatar"</code>: The engine is an avatar.</li>
+ *     </ul>
```
Comment (ctrlaltdavid) [inline libraries/script-engine/src/ScriptEngine.h:128]:
```suggestion
 * @property {string} type - The type of script that is running:
 *     <ul>
 *       <li><code>"client"</code>: An Interface script.</li>
 *       <li><code>"entity_client"</code>: A client entity script.</li>
 *       <li><code>"avatar"</code>: An avatar script.</li>
 *       <li><code>"entity_server"</code>: A server entity script.</li>
 *       <li><code>"agent"</code>: An assignment client script.</li>
 *     </ul>
```

File: libraries/script-engine/src/ScriptEngine.h (hunk 3 above)
Comment (ctrlaltdavid) [reply libraries/script-engine/src/ScriptEngine.h:128]: This originated in the modkit feature, which distinguishes between Interface and avatar script "contexts", whereas the "context" property does not.

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

**Review by ctrlaltdavid — APPROVED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR730/Vircadia-Alpha-PR730-b7d7147_ca80d799.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR730/Vircadia-Alpha-PR730-b7d7147.tar_ca80d799.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR730/Vircadia-Alpha-PR730-b7d7147_ca80d799.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR730/Vircadia-Alpha-PR730-b7d7147.tar_ca80d799.Z

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR730/Vircadia-Alpha-PR730-b7d7147_ca80d799.exe

**Pull request merged**